rovarga/triemap
# gpg

- Remove (mis-)use of Optional
- Mark fields as transient
- Remove Equivalence generic
- [maven-release-plugin] prepare release triemap-1.2.0
- [maven-release-plugin] prepare for next development iteration
- Update shields
- Add LGTM configuration
- Bump odlparent to 8.1.0
- Bump gpg-plugin to 3.0.1
- Bump odlparent to 9.0.5
- Add a Contributing section
- Add SECURITY.md
- Modernize for Java 11
- Adjust LGTM to maven 3.9.1
- Bump odlparent to 10.0.4
- [maven-release-plugin] prepare release triemap-1.2.1
- [maven-release-plugin] prepare for next development iteration
- Create codeql.yml
- Bump versions to 1.3.0-SNAPSHOT
- Update codeql build
- Fixup YAML errors
- Bump odlparent to 12.0.4
- Ditch LGTM
- Enable repository cache in codeql
- Migrate to JUnit5
- Modernize tests
- Seal TrieMap
- Seal AbstractEntrySet
- Seal AbstractKeySet
- Seal TrieSet
- Turn Result into a record
- Update dependency example version
- Document required Java versions
- Remove settings.xml
- Remove deploy/install plugin configuration
- Create maven.yml
- Fix maven build
- Update build status badge
- Disable Update Dependency Graph
- Seal EntryNode
- Rename EntryUtil methods
- Use Mockito in FailedNodeTest
- Use instanceof pattern in AbstractEntrySet.contains()
- Seal BasicNode hierarchy
- Modernize CNode
- Use String.formatted()
- Add @java.io.Serial annotations
- Modernize INode
- Modernize MutableEntrySet
- Improve {Abstract,Mutable}KeySet.iterator() memory overhead
- Eliminate a nested if
- [maven-release-plugin] prepare release triemap-1.3.0
- [maven-release-plugin] prepare for next development iteration
- Bump referece to 1.3.0
- Mark MutableTrieMap.root as transient
- Add sonarcloud integration
- Cache maven repository
- Clean up build triggers
- Cache sonar cache
- Use pull_request_target
- Update badges
- Remove unused configuration
- Remove travis integration
- Rename TrieMap.keySet
- Remove TestInstantiationSpeed
- Refactor assertThrows() tests
- Use an instanceof pattern in CNode
- Verify MainNode is not null
- Improve MutableTrieMap.root
- Share iterator implementation for keySet()
- Remove AbstractKeySet.isEmpty()
- Remove AbstractKeySet.add()
- Remove AbstractEntrySet.map()
- Improve AbstractEntrySet.contains()
- Use generic TrieMap in AbstractEntrySet
- Seal AbstractIterator
- Verify Constants
- Remove an empty line
- Annotate MutableTrieSet.serialVersionUID
- Suppress a SpotBugs warning
- Revert "Use pull_request_target"
- Fix ImmutableTrieSet.addAll()
- Improve Immutable{Entry,Key}Set.removeIf()
- Add TrieSet unit tests
- Move Constants verification
- Fix bundle metadata
- Migrate to use bnd-parent
- Bump maven-gpg-plugin to 3.1.0
